### DISCUSSION TITLE
Checking if release is latest on release created event

### DISCUSSION THREAD

**Discussion by prryplatypus** [community/community#64734] [Actions] [answered]
### Select Topic Area

Question

### Body

Hello,

I've got a workflow that gets dispatched when a new release is created, but I want to know if that release has been marked as the latest or not without having to make an additional API call (see screenshot attached). Is there a way to do this?

![imagen](https://github.com/community/community/assets/25409753/952022d6-634d-4c5a-bd97-9bc412007870)

Our usecase for this would be to know whether tag a Docker image as `latest` or not (if it's just a patch for an LTS version, for example).

TIA.

**Comment by arafajs** [comment]
You can use the` latest_release` variable that is available in all GitHub Actions workflows.

**Reply by prryplatypus** [reply]
Is this documented anywhere? I can't find any information about it.

**Comment by habby1337** [comment]
In action you, for instance, you can access the event payload using the **github** context. You would need to check the value of **release** object's **prerelease** property to see if the release is marked as a pre-release. Additionally, you could check the **release** object's **draft** property to determine if the release is a draft. 

 example:
```
name: Check Latest Release

on:
  release:
    types:
      - created

jobs:
  check-latest-release:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout repository
        uses: actions/checkout@v2

      - name: Determine latest release
        run: |
          latest_release="Latest Release"
          
          if [[ "${{ github.event.release.name }}" == "$latest_release"* ]]; then
            echo "This release is marked as the latest."
          else
            echo "This release is not marked as the latest."
          fi

```

**Reply by prryplatypus** [reply]
This won't work because you're hardcoding the latest release version. I need to get that information directly from GitHub's release information.

**Reply by habby1337** [reply, marked as answer]
oh now i see what you mean:
maybe something on the line of this:
```
 - name: Get release info
        id: release
        run: |
          RELEASE_JSON=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
            "https://api.github.com/repos/${{ github.repository }}/releases/latest")

          LATEST_RELEASE_NAME=$(echo "$RELEASE_JSON" | jq -r '.name')
          IS_LATEST=$(echo "$RELEASE_JSON" | jq -r '.prerelease')

          echo "Latest release name: $LATEST_RELEASE_NAME"
          echo "Is latest release: $IS_LATEST"
```

make sure to have jq tool installed in your workflow env

**Reply by prryplatypus** [reply]
Yup, this is what I'm currently doing. Feels weird that this data is not included in the event payload. I've opened a feature request so hopefully it gets seen and included there. Thanks for your help :)

```bash
latest_tag=$(
            curl -L \
              -H "Accept: application/vnd.github+json" \
              -H "Authorization: Bearer ${{ github.token }}" \
              -H "X-GitHub-Api-Version: 2022-11-28" \
              https://api.github.com/repos/${{ github.repository }}/releases/latest \
              | jq -r '.tag_name'
)

if [[ "${latest_tag}" == "${{ github.event.release.tag_name }}" ]]; then
	...
```

**Reply by habby1337** [reply]
so, I did some more research, apparently this is the best way (currently) to do this kind of verification.
props to you for the feature request!

**Reply by prryplatypus** [reply]
Feature request: https://github.com/orgs/community/discussions/64736
